cartridges/int_bolt_sfra/cartridge/client/default/js/bolt.js

```diff
@@ -108,4 +108,31 @@ exports.initBoltButton = function () {
     }, 100);
 };
 
+/**
+ * This function is to add event listener to order total element if applePay button is detected.
+ * Once change of order total is detected, reload the page to run bolt configure logic again(currently page load is the only way to configure bolt modal)
+ * with updated cart information. This is because applepay only allows sync action, order data needs to be ready(via bolt configure)
+ * before applepay button shows up.
+ * This is a short term solution, once storm side expose 'reconfigure' function. We can reconfigure bolt modal without page reload.
+ */
+exports.addApplePayHandlerIfNeeded = function () {
+    const MAX_COUNT = 30;
+    var timeCounter = 0;
+    var grandTotalClass = '.grand-total';
+    var applePayButtonExist = setInterval(function () {
+        if (timeCounter >= MAX_COUNT) {
```

Comment:
Why the timeout here?

Reply:
Just thinking to save some browser resource, if the apple pay button not showing in 30 seconds, it's probably not a safari browser and apple pay button will never shows, so I think better to stop the timer because it's not necessary.